hpy/tools/autogen/public_api.h

```diff
@@ -115,30 +115,30 @@ HPy_ID(79)
 void HPy_Close(HPyContext *ctx, HPy h);
 
 HPy_ID(80)
-HPy HPyLong_FromLong(HPyContext *ctx, long value);
+HPy HPyLong_FromInt32_t(HPyContext *ctx, int32_t value);
 HPy_ID(81)
-HPy HPyLong_FromUnsignedLong(HPyContext *ctx, unsigned long value);
+HPy HPyLong_FromUInt32_t(HPyContext *ctx, uint32_t value);
 HPy_ID(82)
-HPy HPyLong_FromLongLong(HPyContext *ctx, long long v);
+HPy HPyLong_FromInt64_t(HPyContext *ctx, int64_t v);
 HPy_ID(83)
-HPy HPyLong_FromUnsignedLongLong(HPyContext *ctx, unsigned long long v);
+HPy HPyLong_FromUInt64_t(HPyContext *ctx, uint64_t v);
 HPy_ID(84)
 HPy HPyLong_FromSize_t(HPyContext *ctx, size_t value);
 HPy_ID(85)
 HPy HPyLong_FromSsize_t(HPyContext *ctx, HPy_ssize_t value);
```

Comment:
Maybe unlike CPython we should define `HPy_ssize_t` to be always an alias to, e.g., `uint64_t`?

Reply:
I think we should do this in pieces. Since int/long and ascii/utf-8 are the most common places for confusion, let's just stick with those for this PR.

Reply:
> Maybe unlike CPython we should define `HPy_ssize_t` to be always an alias to, e.g., `uint64_t`?

This would be fine for me but just to consider the original reason for having `Py_ssize_t` in CPython (https://peps.python.org/pep-0353/): As far as I understand, there are two reasons. (1) CPython previously used `int` for indexing which is just too small on 64-bit systems, and (2) because the signed integral type `ssize_t` is not available everywhere.

From my point of view, point (1) implies that the type used for indexing should fit into one register (certainly for performance reasons). Consequently, we should maybe also have such a type but as far as I know, we don't run any tests on 32-bit systems right now. I doubt that HPy even works on 32-bit systems. This indicates that we should probably just remove `ctx_Long_FromSsize_t` from the context and provide an inline helper.

The more interesting part then is: how should we deal with all the other API functions that use `HPy_ssize_t`?
Anyway, this looks like a larger topic which I wouldn't tackle in this PR.